core/includes/file.inc

```diff
@@ -1489,197 +1489,221 @@ function file_space_used($uid = NULL, $status = FILE_STATUS_PERMANENT) {
+  // return the first and only file (delta 0).
+  $multiple_file_upload = is_array($_FILES['files']['name'][$form_field_name]);
+  if (!$multiple_file_upload) {
+    $delta = 0;
+  }
 
-    case UPLOAD_ERR_OK:
-      // Final check that this is a valid upload, if it isn't, use the
-      // default error handler.
-      if (is_uploaded_file($_FILES['files']['tmp_name'][$form_field_name])) {
-        break;
-      }
+  // Return cached objects without processing since the file will have
+  // already been processed and the paths in $_FILES will be invalid.
+  if (isset($upload_cache[$form_field_name])) {
+    if (isset($delta)) {
+      return $upload_cache[$form_field_name][$delta];
+    }
+  }
 
-    // Unknown error
-    default:
-      backdrop_set_message(t('The file %file could not be saved. An unknown error has occurred.', array('%file' => $_FILES['files']['name'][$form_field_name])), 'error');
-      return FALSE;
+  // Prepare uploaded files info so that single and multiple value uploads can
+  // be handled consistently despite differences in the $_FILES array structure.
+  $uploaded_files = $_FILES;
+  if (!$multiple_file_upload) {
+    foreach (array('name', 'type', 'tmp_name', 'error', 'size') as $value)
+    $uploaded_files['files'][$value][$form_field_name] = array($uploaded_files['files'][$value][$form_field_name]);
   }
-  // Begin building file entity.
-  $values = array(
-    'uid' => $user->uid,
-    'status' => 0,
-    'filename' => trim(backdrop_basename($_FILES['files']['name'][$form_field_name]), '.'),
-    'uri' => $_FILES['files']['tmp_name'][$form_field_name],
-    'filesize' => $_FILES['files']['size'][$form_field_name],
-  );
-  $values['filemime'] = file_get_mimetype($values['filename']);
-  $file = new File($values);
 
-  $extensions = '';
-  if (isset($validators['file_validate_extensions'])) {
-    if (isset($validators['file_validate_extensions'][0])) {
-      // Build the list of non-munged extensions if the caller provided them.
-      $extensions = $validators['file_validate_extensions'][0];
+  $files = array();
+  foreach ($uploaded_files['files']['name'][$form_field_name] as $i => $name) {
+    // Check for file upload errors and return FALSE for this file if a lower
+    // level system error occurred. For a complete list of errors:
+    // See http://php.net/manual/features.file-upload.errors.php.
+    switch ($uploaded_files['files']['error'][$form_field_name][$i]) {
+      case UPLOAD_ERR_INI_SIZE:
+      case UPLOAD_ERR_FORM_SIZE:
+        backdrop_set_message(t('The file %file could not be saved because it exceeds %maxsize, the maximum allowed size for uploads.', array('%file' => $name, '%maxsize' => format_size(file_upload_max_size()))), 'error');
+        $files[$i] = FALSE;
+        continue;
+
+      case UPLOAD_ERR_PARTIAL:
+      case UPLOAD_ERR_NO_FILE:
+        backdrop_set_message(t('The file %file could not be saved because the upload did not complete.', array('%file' => $name)), 'error');
```

Comment:
```suggestion
        backdrop_set_message(t('The file %file could not be saved, because the upload did not complete.', array('%file' => $name)), 'error');
```